src/pages/exchange-crypto/exchange-crypto.ts

```diff
@@ -66,217 +103,508 @@ export class ExchangeCryptoPage {
-          this.showErrorAndBack(
-            null,
-            this.translate.instant(
-              'Changelly is not available at this moment. Please, try again later.'
-            )
-          );
-          return;
+    this.exchangeCryptoProvider.getSwapTxs().then(res => {
+      this.oneInchSwapTxs = res.oneInchSwapTxs;
+    });
+  }
+
+  ngOnDestroy() {
+    if (this.timeout) clearTimeout(this.timeout);
+  }
+
+  ionViewDidLoad() {
+    this.logger.info('Loaded: ExchangeCryptoPage');
+
+    this.getExchangesCurrencies();
+  }
+
+  private async getExchangesCurrencies() {
+    const reflect = promiseObj => {
+      return promiseObj.promise.then(
+        v => {
+          return { exchange: promiseObj.exchange, status: 'ok', data: v };
+        },
+        error => {
+          return {
+            exchange: promiseObj.exchange,
+            status: 'failed',
+            reason: error
+          };
         }
+      );
+    };
 
-        if (
-          data &&
-          data.result &&
-          _.isArray(data.result) &&
-          data.result.length > 0
-        ) {
-          this.supportedCoins = _.intersection(
-            this.currencyProvider.getAvailableCoins(),
-            data.result
-          );
-          const coinsToRemove = ['xrp', 'busd'];
-          coinsToRemove.forEach((coin: string) => {
-            const index = this.supportedCoins.indexOf(coin);
-            if (index > -1) {
-              this.logger.debug(
-                `Removing ${coin.toUpperCase()} from supported coins`
+    const promises = [
+      {
+        exchange: 'changelly',
+        promise: this.changellyProvider.getCurrencies()
+      },
+      { exchange: '1inch', promise: this.oneInchProvider.getCurrencies1inch() }
+    ];
+    const results = await Promise.all(promises.map(reflect));
+    const successfulPromises = results.filter(p => p.status === 'ok');
+    const failedPromises = results.filter(p => p.status === 'failed');
+
+    failedPromises.forEach(promise => {
+      switch (promise.exchange) {
+        case '1inch':
+          this.logger.error('1Inch getCurrencies Error');
+          if (promise.reason && promise.reason.message)
+            this.logger.error(promise.reason.message);
+          break;
+
+        case 'changelly':
+          this.logger.error('Changelly getCurrencies Error');
+          if (promise.reason && promise.reason.message)
+            this.logger.error(promise.reason.message);
+          break;
+
+        default:
+          break;
+      }
+    });
+
+    successfulPromises.forEach(promise => {
+      switch (promise.exchange) {
+        case '1inch':
+          if (!promise.data || !promise.data.tokens) {
+            this.logger.error('1Inch getCurrencies Error');
+            // this.showErrorAndBack(
+            //   null,
+            //   this.translate.instant(
+            //     '1Inch is not available at this moment. Please, try again later.'
+            //   )
+            // );
```

Comment:
what about this error?

Reply:
This error handle is no longer necessary. I removed it